src/TuDa.CIMS.Api/Repositories/WorkingGroupRepository.cs

```diff
@@ -0,0 +1,138 @@
+﻿using Microsoft.EntityFrameworkCore;
+using TuDa.CIMS.Api.Database;
+using TuDa.CIMS.Api.Interfaces;
+using TuDa.CIMS.Shared.Dtos;
+using TuDa.CIMS.Shared.Entities;
+
+namespace TuDa.CIMS.Api.Repositories;
+
+public class WorkingGroupRepository : IWorkingGroupRepository
+{
+    private readonly CIMSDbContext _context;
+
+    public WorkingGroupRepository(CIMSDbContext context)
+    {
+        _context = context;
+    }
+
+    /// <summary>
+    /// Returns an existing Working Group with the specific id.
+    /// </summary>
+    /// <param name="id">the specific id of the Working Group</param>
+    /// <returns></returns>
+    public async Task<WorkingGroup?> GetOneAsync(Guid id)
+    {
+        return await _context.WorkingGroups.Where(i => i.Id == id).SingleOrDefaultAsync();
+    }
+
+    /// <summary>
+    /// Returns all existing Working Groups of the database.
+    /// </summary>
+    public async Task<IEnumerable<WorkingGroup>> GetAllAsync()
+    {
+        return await _context.WorkingGroups.Include(workingGroup => workingGroup.Professor)
+            .Include(workingGroup => workingGroup.Students).Include(workingGroup => workingGroup.Purchases)
+            .ToListAsync();
+    }
+
+
+    /// <summary>
+    /// Updates an existing Working Group with the specified id using the provided update model.
+    /// </summary>
+    /// <param name="id">the specific id of the Working Group</param>
+    /// <param name="updateModel">the model containing the updated values for the Working Group </param>
+    public async Task<ErrorOr<Updated>> UpdateAsync(Guid id, UpdateWorkingGroupDto updateModel)
+    {
+        var existingItem = await _context
+            .WorkingGroups.Include(workingGroup => workingGroup.Professor)
+            .Include(workingGroup => workingGroup.Students).Include(workingGroup => workingGroup.Purchases)
+            .SingleOrDefaultAsync(i => i.Id == id);
+
+        if (existingItem is null)
+        {
+            return Error.NotFound("WorkingGroups.update", $"Working Group with id {id} was not found.");
+        }
+
+        existingItem.Professor = updateModel.Professor;
+        existingItem.Students = updateModel.Students;
+        existingItem.PhoneNumber = updateModel.PhoneNumber;
+        existingItem.Purchases = updateModel.Purchases;
```

Comment:
🌱 We could think about updating the students through another endpoint. Because now we have to provide the full list without something or with something added to not delete the rest.
We could also add a `RemoveStudentsList` and `AddStudentsList` to the `UpdateModel`

Reply:
Would that be sufficient to not add new endpoints by adding this to `UpdateAsync`

``` 
// Handle adding new students
    if (updateModel.AddStudentsList != null && updateModel.AddStudentsList.Any() )
    {
        foreach (var student in updateModel.AddStudentsList)
        {
            // Ensure the student exists in the database before adding
            if (!_context.Students.Any(s => s.Id == student.Id))
            {
                _context.Students.Add(student);
            }
        }
        existingItem.Students.AddRange(updateModel.AddStudentsList);
    }

// Handle removing students 
    if (updateModel.RemoveStudentsList != null && updateModel.RemoveStudentsList.Any() )
    {
        foreach (var student in updateModel.RemoveStudentsList)
        {
            var studentToRemove = existingItem.Students.FirstOrDefault(s => s.Id == student.Id);
            if (studentToRemove != null)
            {
                existingItem.Students.Remove(studentToRemove);
            }
        }
    }

    await _context.SaveChangesAsync();
    return Result.Updated;}
```

Reply:
Yeah, that would work. But let's talk tomorrow about it